src/status_im/contexts/profile/settings/screens/password/change_password/effects.cljs

```diff
@@ -0,0 +1,18 @@
+(ns status-im.contexts.profile.settings.screens.password.change-password.effects
+  (:require [native-module.core :as native-module]
+            [promesa.core :as promesa]
+            [status-im.common.keychain.events :as keychain]
+            [utils.re-frame :as rf]
+            [utils.security.core :as security]))
+
+(rf/reg-fx
+ :effects.change-password/change-password
```

Comment:
Just wanted to share that I like the style added by @clauxx in this case, I feel it more comfortable. @ilmotta  Do we really want to add a strict code style for this case?

Reply:
@ulisesmac, tbh I think either style is fine. Really, the only reason I prefer the other way is because of its practical benefit (at least to me, a few times every day).

I have a theory as to why our repo uses the IDs on the second line. By default, text editors will format arguments to s-exps spanning over multiple lines aligned with the function name. This adds unnecessary indentation that's undesirable in some cases. For example, like this:

```clojure
(rf/reg-event-fx :change-password/submit
                 (fn [{:keys [db]}]
                   ...))
``` 

The way Clojure devs usually deal with this problem without a specific formatting rule from an external too is to either add a line break after `reg-event-fx` (what we did and still do) or, when the team is small, have devs add a formatting rule in their editor (easy in Emacs).

But we have zprint for a while and we already added the rules to support this style in `zprintrc`.

> Do we really want to add a strict code style for this case?

Style rules are always questionable. For this one I think yes, it literally simplifies grepping to find definition and/or usages. But as I said, I don't know if it's possible to have a rule for this because we have in our `zprintrc` that we respect line breaks as much as possible, hence if the dev puts the ID on a separate line zprint won't complaint.

Reply:
+1 for leaving the event names on the same line. For the same reason that Icaro stated.

Reply:
For me it's easy to configure the code style of `rf/reg-event-fx` locally, but I still prefer the two line version.

idk, probably this should be voted.

Personally, I'm used to the two line style and I like it too much. same for re-frame subs. BTW what is the a reason behind you don't keep `reg-sub` and the id in the same line? why is it only applied to `reg-event-fx` and `reg-fx`? @vkjr @ilmotta

Reply:
@ulisesmac, maybe my initial message was confusing. I wrote `reg-*` to mean all re-frame registration functions receive IDs, and those can always go on the same line as the reg function. This includes `reg-sub` for sure.

Just like the Clojure style guide, what @vkjr and I are advocating is based on a practical advantage. In terms of aesthetics both are fine by me.

Look how useless a line grep looks as of today. There's no easy way to find a keyword that's a `reg-sub` (the same problem applies to other re-frame registration calls).

<img src="https://github.com/status-im/status-mobile/assets/46027/06a3cce5-9e7f-4d32-bcd4-b26e5bc85e07" width="400" />

Now look what happens when the IDs are written on the same line. I can easily filter and find the definition of any re-frame registration call (if our entire codebase was respecting this style).
 
<img src="https://github.com/status-im/status-mobile/assets/46027/e16f0474-da96-4dbd-bf1e-f0e15bed2401" width="400" />

Reply:
> idk, probably this should be voted.

I agree with you. In this thread it's just a few of us, but it's been useful because it's the first time I remember we talk about this. I think Emacs users are suffering more from this style choice due to how grep results are presented line by line in the minibuffer.

Reply:
I would also vote for this cause it's been a bit of a guessing game finding event/sub definitions, but I think this should be enforced by the formatter to get consistent results.

Reply:
The inconvenience also exists in VSCode, not only in emacs:

![Screenshot 2024-04-10 at 15 19 44](https://github.com/status-im/status-mobile/assets/5786310/4c89a202-7a69-4ee2-89ca-c86797ae65c1)

Reply:
> I think this should be enforced by the formatter to get consistent results.

Well, let's summon @yqrashawn for zprint stuff! My rough guess is that it can be tricky.

@yqrashawn, do you know if it's possible to tell zprint to prefer to keep re-frame IDs (keywords passed to `reg-sub`, `reg-event-fx`, etc) in the same line as the function call while we still use `:respect-nl` in the config?

Reply:
@ilmotta OK, got it.

I honestly haven't looked for `reg-*`, it's just not part of my workflow, I usually look for a keyword (the event/sub ID) being used in a `dispatch`, reg-*, `sub`, etc, and I get all the results related:

![image](https://github.com/status-im/status-mobile/assets/90291778/ee746749-c058-4192-bc04-483fdff5ddee)


But since you'll have a benefit from it, either is fine for me.

Reply:
logged here https://github.com/status-im/status-mobile/issues/17641#issue-1943865259
I'll look into it, see if it's possible

I look for keyword instead of reg-xxx as well
I tried to implement something like find definition for rf event keys in emacs (based on ripgrep) before but failed
I just use my emacs default ripgrep setup for now